tests/checkpointing/test_model_checkpoint.py

```diff
@@ -498,6 +498,20 @@ def test_none_monitor_top_k(tmpdir):
     ModelCheckpoint(dirpath=tmpdir, save_top_k=0)
 
 
+def test_invalid_every_n_epoch(tmpdir):
+    """ Test different configurations for every_n_epochs. """
+    ModelCheckpoint(dirpath=tmpdir, every_n_epochs=0, every_n_batches=1)
+    ModelCheckpoint(dirpath=tmpdir, every_n_epochs=3)
+    ModelCheckpoint(dirpath=tmpdir, every_n_epochs=-1, every_n_batches=1)
+
+
+def test_invalid_every_n_batches(tmpdir):
+    """ Test different configurations for every_n_batches. """
+    ModelCheckpoint(dirpath=tmpdir, every_n_batches=0)
+    ModelCheckpoint(dirpath=tmpdir, every_n_batches=3)
+    ModelCheckpoint(dirpath=tmpdir, every_n_batches=-1, every_n_epochs=2)
```

Comment:
Add `with pytest.raises` where appropriate

Reply:
i've removed the restrictions on every_n_batches/every_n_epochs being < 0. instead of failing, we ignore those values, so this test is less maningful. what do you think?

Reply:
I personally prefer to have it restricted and avoid possible confusion